File: samples/shared-element-samples/simple-stack-example-sharedelement-fragment/src/main/java/com/example/fragmenttransitions/core/navigation/BaseFragment.java
package com.example.fragmenttransitions.core.navigation;

import android.os.Bundle;

import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.view.View;

import com.zhuinden.simplestackextensions.fragments.KeyedFragment;

/**
 * Created by Zhuinden on 2020. 12. 18..
 */

public class BaseFragment
        extends KeyedFragment {
    public BaseFragment() {
    }

    public BaseFragment(@LayoutRes int layoutRes) {
        super(layoutRes);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        startPostponedEnterTransition();
    }
}

File: samples/legacy-samples/simple-stack-example-multistack-fragment/src/main/java/com/zhuinden/simplestackdemomultistack/core/navigation/BaseFragment.kt
package com.zhuinden.simplestackdemomultistack.core.navigation

import androidx.annotation.LayoutRes
import androidx.fragment.app.Fragment

/**
 * Created by Zhuinden on 2018.09.17.
 */

open class BaseFragment(@LayoutRes layoutId: Int) : Fragment(layoutId) {
    fun <T : MultistackFragmentKey> getKey(): T = requireArguments().getParcelable<T>("KEY")!!
}
